### ISSUE TITLE
BINARY_FLOAT return nil (Rails 2.4 with oracle-enhanced 1.6.8)

### ISSUE THREAD

**Issue by shozayen** [rsim/oracle-enhanced#1244]
I am using Rails 4.2.7 with oracle adapter 1.6.8. I can save the data into oracle DB as float without any issue but when I retrieve the data, the data in Float columns returned as **nil**. 

**Image:**  
[https://i.stack.imgur.com/yA2GI.png](https://i.stack.imgur.com/yA2GI.png)

**Comment by yahonda** [comment]
Thanks for your report. Would you create a executable test case based on this template below.

```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'rails', '~> 4.2.0'
  gem 'activerecord-oracle_enhanced-adapter', '~> 1.6.0'
  gem 'ruby-oci8'
  gem 'minitest'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD
}

ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
  end

  create_table :comments, :force => true do |t|
    t.integer :post_id
  end
end

class Post < ActiveRecord::Base
  has_many :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post
end

class BugTest < Minitest::Test
  def test_association_stuff
    post = Post.create!
    post.comments << Comment.create!

    assert_equal 1, post.comments.count
    assert_equal 1, Comment.count
    assert_equal post.id, Comment.first.post.id
  end
end
```

Thanks.

**Comment by shozayen** [comment]
I have a problem installing **ruby-oci8** gem to conduct the test. I am using Jruby 9.1.8 x64. 


```

RuntimeError: RuntimeError
     get at C:/jruby-9.1.8.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.3/ext/oci8/oraconf.rb:345
  <main> at extconf.rb:22
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=C:/jruby-9.1.8.0/bin/jruby
        --with-instant-client
        --without-instant-client

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  C:/jruby-9.1.8.0/lib/ruby/gems/shared/extensions/universal-java-1.8/2.3.0/ruby-oci8-2.2.3/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in C:/jruby-9.1.8.0/lib/ruby/gems/shared/gems/ruby-oci8-2.2.3 for inspection.
Results logged to C:/jruby-9.1.8.0/lib/ruby/gems/shared/extensions/universal-java-1.8/2.3.0/ruby-oci8-2.2.3/gem_make.out
```

**Comment by koic** [comment]
ruby-oci8 gem is not needed in JRuby.

Would you remove the following code and try again, please?

```ruby
gem "ruby-oci8"
```

**Comment by shozayen** [comment]
Hi,

The 3 test cases run successfully but you didn't add any float for test, check below code : 


```

Fetching gem metadata from https://rubygems.org/..........                                                              Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies.....                                                                                             Using rake 12.0.0 (was 10.5.0)
Using i18n 0.8.1 (was 0.8.0)                                                                                            Using minitest 5.10.1
Using thread_safe 0.3.6 (java) (was 0.3.5)
Using builder 3.2.3
Using erubis 2.7.0
Using nokogiri 1.7.1 (java) (was 1.7.0.1)
Using rack 1.6.5
Using mime-types-data 3.2016.0521
Using arel 6.0.4
Using activerecord-oracle_enhanced-adapter 1.6.8 (was 1.6.7)
Using bundler 1.14.6
Using concurrent-ruby 1.0.5 (java) (was 1.0.4)                                                                          Using thor 0.19.4
Using tzinfo 1.2.2                                                                                                      Using loofah 2.0.3
Using rack-test 0.6.3                                                                                                   Using mime-types 3.1 (was 2.99.3)
Using sprockets 3.7.1 (was 2.12.4)
Installing activesupport 4.2.8 (was 4.2.7)                                                                              Using rails-html-sanitizer 1.0.3
Using mail 2.6.4                                                                                                        Using rails-deprecated_sanitizer 1.0.3
Using globalid 0.3.7                                                                                                    Installing activemodel 4.2.8 (was 4.2.7)
Using rails-dom-testing 1.0.8                                                                                           Installing activejob 4.2.8 (was 4.2.7)
Installing activerecord 4.2.8 (was 4.2.7)                                                                               Installing actionview 4.2.8 (was 4.2.7)
Installing actionpack 4.2.8 (was 4.2.7)                                                                                 Installing actionmailer 4.2.8 (was 4.2.7)
Installing railties 4.2.8 (was 4.2.7)                                                                                   Using sprockets-rails 3.2.0 (was 2.3.3)
Installing rails 4.2.8 (was 4.2.7)                                                                                      -- create_table(:posts, {:force=>true})
D, [2017-03-21T14:02:28.378000 #17732] DEBUG -- :   [1m[36m (769.0ms)[0m  [1mCREATE TABLE "POSTS" ("ID" NUMBER(38) NOT NULL PRIMARY KEY) [0m
D, [2017-03-21T14:02:28.574000 #17732] DEBUG -- :   [1m[35m (173.0ms)[0m  CREATE SEQUENCE "POSTS_SEQ" START WITH 10000
   -> 2.7460s                                                                                                           -- create_table(:coments, {:force=>true})
D, [2017-03-21T14:02:28.883000 #17732] DEBUG -- :   [1m[36m (254.0ms)[0m  [1mCREATE TABLE "COMENTS" ("ID" NUMBER(38) NOT NULL PRIMARY KEY, "POST_ID" NUMBER(38)) [0m
D, [2017-03-21T14:02:28.964000 #17732] DEBUG -- :   [1m[35m (72.0ms)[0m  CREATE SEQUENCE "COMENTS_SEQ" START WITH 10000
   -> 0.3782s
Run options: --seed 37401

# Running:
                                                                                                                        D, [2017-03-21T14:02:29.622000 #17732] DEBUG -- :   [1m[36m (168.0ms)[0m  [1mSELECT column_name AS name, data_type AS sql_type, data_default, nullable, virtual_column, hidden_column, data_type_owner AS sql_type_owner, DECODE(data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale FROM all_tab_cols WHERE owner = 'SHIKO_DEV' AND table_name = 'POSTS' AND hidden_column = 'NO' ORDER BY column_id[0m                                                             D, [2017-03-21T14:02:29.795000 #17732] DEBUG -- :   [1m[35m (154.0ms)[0m  SELECT DECODE(table_name, UPPER(table_name), LOWER(table_name), table_name) FROM all_tables WHERE owner = SYS_CONTEXT('userenv', 'current_schema') AND secondary = 'N'
D, [2017-03-21T14:02:29.879000 #17732] DEBUG -- :   [1m[36mSequence (48.0ms)[0m  [1mselect us.sequence_name from all_sequences us where us.sequence_owner = 'SHIKO_DEV' and us.sequence_name = 'POSTS_SEQ'[0m                          D, [2017-03-21T14:02:31.014000 #17732] DEBUG -- :   [1m[35mPrimary Key (1127.0ms)[0m  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'SHIKO_DEV' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name                                                     D, [2017-03-21T14:02:31.143000 #17732] DEBUG -- :   [1m[36mSequence (10.0ms)[0m  [1mselect us.sequence_name from all_sequences us where us.sequence_owner = 'SHIKO_DEV' and us.sequence_name = 'POSTS_SEQ'[0m                          D, [2017-03-21T14:02:31.160000 #17732] DEBUG -- :   [1m[35mPrimary Key (11.0ms)[0m  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'SHIKO_DEV' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name                                                       D, [2017-03-21T14:02:31.464000 #17732] DEBUG -- :   [1m[36mPrimary Key Trigger (289.0ms)[0m  [1m          SELECT trigger_name                                                                                                                         FROM all_triggers                                                                                                       WHERE owner = 'SHIKO_DEV'                                                                                             AND trigger_name = 'POSTS_PKT'                                                                                          AND table_owner = 'SHIKO_DEV'                                                                                       AND table_name = 'POSTS'                                                                                                AND status = 'ENABLED'                                                                                      [0m                                                                                                                    D, [2017-03-21T14:02:31.525000 #17732] DEBUG -- :   [1m[35mSQL (21.0ms)[0m  INSERT INTO "POSTS" ("ID") VALUES (:a1)  [["id", 10000]]
D, [2017-03-21T14:02:31.635000 #17732] DEBUG -- :   [1m[36m (29.0ms)[0m  [1mSELECT column_name AS name, data_type AS sql_type, data_default, nullable, virtual_column, hidden_column, data_type_owner AS sql_type_owner, DECODE(data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale FROM all_tab_cols WHERE owner = 'SHIKO_DEV' AND table_name = 'COMENTS' AND hidden_column = 'NO' ORDER BY column_id[0m
D, [2017-03-21T14:02:31.675000 #17732] DEBUG -- :   [1m[35mSequence (9.0ms)[0m  select us.sequence_name from all_sequences us where us.sequence_owner = 'SHIKO_DEV' and us.sequence_name = 'COMENTS_SEQ'
D, [2017-03-21T14:02:31.687000 #17732] DEBUG -- :   [1m[36mPrimary Key (8.0ms)[0m  [1mSELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'SHIKO_DEV' AND c.table_name = 'COMENTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name[0m
D, [2017-03-21T14:02:31.731000 #17732] DEBUG -- :   [1m[35mSequence (9.0ms)[0m  select us.sequence_name from all_sequences us where us.sequence_owner = 'SHIKO_DEV' and us.sequence_name = 'COMENTS_SEQ'
D, [2017-03-21T14:02:31.749000 #17732] DEBUG -- :   [1m[36mPrimary Key (12.0ms)[0m  [1mSELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'SHIKO_DEV' AND c.table_name = 'COMENTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name[0m
D, [2017-03-21T14:02:31.763000 #17732] DEBUG -- :   [1m[35mPrimary Key Trigger (9.0ms)[0m            SELECT trigger_name
          FROM all_triggers
          WHERE owner = 'SHIKO_DEV'
            AND trigger_name = 'COMENTS_PKT'
            AND table_owner = 'SHIKO_DEV'
            AND table_name = 'COMENTS'
            AND status = 'ENABLED'

D, [2017-03-21T14:02:31.788000 #17732] DEBUG -- :   [1m[36mSQL (8.0ms)[0m  [1mINSERT INTO "COMENTS" ("ID") VALUES (:a1)[0m  [["id", 10000]]
D, [2017-03-21T14:02:31.844000 #17732] DEBUG -- :   [1m[35mSQL (10.0ms)[0m  UPDATE "COMENTS" SET "POST_ID" = :a1 WHERE "COMENTS"."ID" = :a2  [["post_id", 10000], ["id", 10000]]
D, [2017-03-21T14:02:31.880000 #17732] DEBUG -- :   [1m[36m (12.0ms)[0m  [1mSELECT COUNT(*) FROM "COMENTS" WHERE "COMENTS"."POST_ID" = :a1[0m  [["post_id", 10000]]
D, [2017-03-21T14:02:31.898000 #17732] DEBUG -- :   [1m[35m (11.0ms)[0m  SELECT COUNT(*) FROM "COMENTS"
D, [2017-03-21T14:02:31.921000 #17732] DEBUG -- :   [1m[36mComent Load (10.0ms)[0m  [1mSELECT * FROM (SELECT  "COMENTS".* FROM "COMENTS"  ORDER BY "COMENTS"."ID" ASC ) WHERE ROWNUM <= 1[0m
D, [2017-03-21T14:02:32.052000 #17732] DEBUG -- :   [1m[35mPost Load (10.0ms)[0m  SELECT  "POSTS".* FROM "POSTS" WHERE "POSTS"."ID" = :a1 AND ROWNUM <= 1   [["id", 10000]]
.

Finished in 2.711792s, 0.3688 runs/s, 1.1063 assertions/s.

1 runs, 3 assertions, 0 failures, 0 errors, 0 skips

```

**Comment by shozayen** [comment]
I've added a float column to Post model as below and entered float data is returned as nil.


```
  create_table :posts, :force => true do |t|
    t.float :stars
  end
```

Add line to test the float
```
class BugTest < Minitest::Test
  def test_association_stuff
    ps = {stars: 4.4}
    post = Post.create(ps)
    post.coments << Coment.create!
    
    assert_equal 1, post.coments.count
    assert_equal 1, Coment.count
    assert_equal post.id, Coment.first.post.id

    puts Post.first.as_json
    
  end
end
```

Result
```

[2017-03-21T17:07:56.922000 #15220] DEBUG -- :   [1m[36mPost Load (10.0ms)[0m  [1mSELECT * FROM (SELECT  "POSTS".* FROM "POSTS"  ORDER BY "POSTS"."ID" ASC ) WHERE ROWNUM <= 1[0m
{"id"=>10000, "stars"=>nil}
.

Finished in 1.030406s, 0.9705 runs/s, 3.8820 assertions/s.

1 runs, 4 assertions, 0 failures, 0 errors, 0 skips

```

**Comment by yahonda** [comment]
Thanks for the update. I think I am also able to produce it. Also reproduced with Rails 5.1.0.rc1 and Oracle enhanced adapter 1.8.0.rc1.

* jruby 9.1.8.0 (2.3.1) 2017-03-06 90fc7ab OpenJDK 64-Bit Server VM 25.121-b14 on 1.8.0_121-b14 +jit [linux-x86_64]
* Oracle Database 12c 12.2.0.1

```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'rails', '~> 5.1.0.rc'
  gem 'activerecord-oracle_enhanced-adapter', '~> 1.8.0.rc'
  gem 'minitest'
  platforms :ruby do
    gem "ruby-oci8"
  end
  gem 'pry'
  gem 'pry-nav'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD
}

ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
    t.float :stars
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_float
    ps = {stars: 4.4}
    post = Post.create(ps)
    assert_equal 4.4, post.stars
    post.save
    assert_equal 4.4, post.stars
    post.reload
    assert_equal 4.4, post.stars
  end
end
```

**Comment by yahonda** [comment]
```ruby
$ ruby rep51.rb
The dependency ruby-oci8 (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for java but the dependency is only for ruby. To add those platforms to the bundle, run `bundle lock --add-platform ruby`.
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies................
Using rake 12.0.0
Using concurrent-ruby 1.0.5 (java)
Using i18n 0.8.1
Using minitest 5.10.1
Using thread_safe 0.3.6 (java)
Using builder 3.2.3
Using erubi 1.6.0
Using nokogiri 1.7.1 (java)
Using rack 2.0.1
Using nio4r 2.0.0 (java)
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 8.0.0
Using ruby-plsql 0.6.0
Using bundler 1.14.6
Using coderay 1.1.1
Using ffi 1.9.18 (java)
Using method_source 0.8.2
Using slop 3.6.0
Using thor 0.19.4
Using tzinfo 1.2.2
Using loofah 2.0.3
Using rack-test 0.6.3
Using sprockets 3.7.1
Using websocket-driver 0.6.5 (java)
Using mime-types 3.1
Using spoon 0.0.6
Using activesupport 5.1.0.rc1
Using rails-html-sanitizer 1.0.3
Using mail 2.6.4
Using pry 0.10.4 (java)
Using rails-dom-testing 2.0.2
Using globalid 0.3.7
Using activemodel 5.1.0.rc1
Using pry-nav 0.2.4
Using actionview 5.1.0.rc1
Using activejob 5.1.0.rc1
Using activerecord 5.1.0.rc1
Using actionpack 5.1.0.rc1
Using activerecord-oracle_enhanced-adapter 1.8.0.rc1
Using actioncable 5.1.0.rc1
Using actionmailer 5.1.0.rc1
Using railties 5.1.0.rc1
Using sprockets-rails 3.2.0
Using rails 5.1.0.rc1
-- create_table(:posts, {:force=>true})
D, [2017-03-22T18:32:05.439623 #12041] DEBUG -- :    (51.1ms)  DROP TABLE "POSTS"
D, [2017-03-22T18:32:05.465932 #12041] DEBUG -- :    (19.0ms)  DROP SEQUENCE "POSTS_SEQ"
D, [2017-03-22T18:32:05.508431 #12041] DEBUG -- :    (33.3ms)  CREATE TABLE "POSTS" ("ID" NUMBER(38) NOT NULL PRIMARY KEY, "STARS" BINARY_FLOAT)
D, [2017-03-22T18:32:05.517064 #12041] DEBUG -- :    (6.3ms)  CREATE SEQUENCE "POSTS_SEQ" START WITH 10000
   -> 1.6893s
D, [2017-03-22T18:32:05.617826 #12041] DEBUG -- :    (9.8ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
D, [2017-03-22T18:32:06.131660 #12041] DEBUG -- :    (30.2ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'AR_INTERNAL_METADATA' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-03-22T18:32:06.208197 #12041] DEBUG -- :   ActiveRecord::InternalMetadata Load (18.9ms)  SELECT  "AR_INTERNAL_METADATA".* FROM "AR_INTERNAL_METADATA" WHERE "AR_INTERNAL_METADATA"."KEY" = :a1 FETCH FIRST :a2 ROWS ONLY  [["key", "environment"], ["LIMIT", 1]]
D, [2017-03-22T18:32:06.219931 #12041] DEBUG -- :    (2.9ms)  SELECT SYS_CONTEXT('userenv', 'current_schema') FROM dual
Run options: --seed 25482

# Running:

D, [2017-03-22T18:32:06.950138 #12041] DEBUG -- :    (14.1ms)  SELECT cols.column_name AS name, cols.data_type AS sql_type, cols.data_default, cols.nullable, cols.virtual_column, cols.hidden_column, cols.data_type_owner AS sql_type_owner, DECODE(cols.data_type, 'NUMBER', data_precision, 'FLOAT', data_precision, 'VARCHAR2', DECODE(char_used, 'C', char_length, data_length), 'RAW', DECODE(char_used, 'C', char_length, data_length), 'CHAR', DECODE(char_used, 'C', char_length, data_length), NULL) AS limit, DECODE(data_type, 'NUMBER', data_scale, NULL) AS scale, comments.comments as column_comment FROM all_tab_cols cols, all_col_comments comments WHERE cols.owner = 'ORACLE_ENHANCED' AND cols.table_name = 'POSTS' AND cols.hidden_column = 'NO' AND cols.owner = comments.owner AND cols.table_name = comments.table_name AND cols.column_name = comments.column_name ORDER BY cols.column_id
D, [2017-03-22T18:32:07.063381 #12041] DEBUG -- :    (80.5ms)  SELECT LOWER(view_name) FROM all_views WHERE owner = SYS_CONTEXT('userenv', 'current_schema')
D, [2017-03-22T18:32:07.602667 #12041] DEBUG -- :   Sequence (2.9ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-03-22T18:32:07.610534 #12041] DEBUG -- :   Primary Key (6.0ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-03-22T18:32:08.143095 #12041] DEBUG -- :   Sequence (3.1ms)  select us.sequence_name from all_sequences us where us.sequence_owner = 'ORACLE_ENHANCED' and us.sequence_name = upper('POSTS_seq')
D, [2017-03-22T18:32:08.147083 #12041] DEBUG -- :   Primary Key (2.5ms)  SELECT cc.column_name FROM all_constraints c, all_cons_columns cc WHERE c.owner = 'ORACLE_ENHANCED' AND c.table_name = 'POSTS' AND c.constraint_type = 'P' AND cc.owner = c.owner AND cc.constraint_name = c.constraint_name
D, [2017-03-22T18:32:08.150657 #12041] DEBUG -- :   Primary Key Trigger (2.2ms)            SELECT trigger_name
          FROM all_triggers
          WHERE owner = 'ORACLE_ENHANCED'
            AND trigger_name = q'[POSTS_PKT]'
            AND table_owner = 'ORACLE_ENHANCED'
            AND table_name = q'[POSTS]'
            AND status = 'ENABLED'

D, [2017-03-22T18:32:08.186912 #12041] DEBUG -- :   SQL (18.1ms)  INSERT INTO "POSTS" ("STARS", "ID") VALUES (:a1, :a2)  [["stars", 4.4], ["id", 10000]]
D, [2017-03-22T18:32:08.220333 #12041] DEBUG -- :   Post Load (11.9ms)  SELECT  "POSTS".* FROM "POSTS" WHERE "POSTS"."ID" = :a1 FETCH FIRST :a2 ROWS ONLY  [["id", 10000], ["LIMIT", 1]]
F

Finished in 1.866434s, 0.5358 runs/s, 1.6073 assertions/s.

  1) Failure:
BugTest#test_float [rep51.rb:66]:
Expected: 4.4
  Actual: nil

1 runs, 3 assertions, 1 failures, 0 errors, 0 skips
$
```

**Comment by shozayen** [comment]
@yahonda 
Good news that you able to reproduce at your own side, I appreciate any updates regarding the fixes.

**Comment by jhadeepakkumar14** [comment]
This will work. I tried saving float value as decimal with proper precision and scale. After getting decimal value we can call to_f to convert that in float if required. This way we can save and fetch float value from oracle db using this adapter.

```
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'rails', '~> 5.1.0.rc'
  gem 'activerecord-oracle_enhanced-adapter', '~> 1.8.0.rc'
  gem 'minitest'
  platforms :ruby do
    gem "ruby-oci8"
  end
  gem 'pry'
  gem 'pry-nav'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'
require 'active_record/connection_adapters/oracle_enhanced_adapter'

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# Set Oracle enhanced adapter specific connection parameters
DATABASE_NAME = ENV['DATABASE_NAME'] || 'orcl'
DATABASE_HOST = ENV['DATABASE_HOST']
DATABASE_PORT = ENV['DATABASE_PORT']
DATABASE_USER = ENV['DATABASE_USER'] || 'oracle_enhanced'
DATABASE_PASSWORD = ENV['DATABASE_PASSWORD'] || 'oracle_enhanced'
DATABASE_SYS_PASSWORD = ENV['DATABASE_SYS_PASSWORD'] || 'admin'

CONNECTION_PARAMS = {
  :adapter => "oracle_enhanced",
  :database => DATABASE_NAME,
  :host => DATABASE_HOST,
  :port => DATABASE_PORT,
  :username => DATABASE_USER,
  :password => DATABASE_PASSWORD
}

ActiveRecord::Base.establish_connection(CONNECTION_PARAMS)

ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, :force => true do |t|
    #precision and scale can be anything
    t.decimal :stars, precision: 6, scale: 3 
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_float
    ps = {stars: 4.4}
    post = Post.create(ps)
    assert_equal 4.4, post.stars
    post.save
    assert_equal 4.4, post.stars
    post.reload
    assert_equal 4.4, post.stars
  end
end
```

**Comment by yahonda** [comment]
Closed as #1254 fixes it. Will back port this fix and release Oracle enhanced adapter 1.6.9 and 1.7.11.
Thanks for using Oracle enhanced adapter and reporting with reproduce case.

**Comment by koic** [comment]
Oracle enhanced adapter 1.6.9 and 1.7.11 has been released. Thanks.
